### PR TITLE
Support for Tower Workflow Job.

### PR DESCRIPTION
**Pull request by lfu**
Depends on https://github.com/ManageIQ/manageiq/pull/17654.
Depends on https://github.com/ansible/ansible_tower_client_ruby/pull/111.
Blocks https://github.com/ManageIQ/manageiq-ui-classic/pull/4157.

https://bugzilla.redhat.com/show_bug.cgi?id=1590975

@miq-bot add_label enhancement

### CODE REVIEW

File: spec/models/manageiq/providers/ansible_tower/automation_manager/workflow_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+describe ManageIQ::Providers::AnsibleTower::AutomationManager::WorkflowJob do
+  it_behaves_like 'ansible workflow job'
```
Comment (bzwei) [inline spec/models/manageiq/providers/ansible_tower/automation_manager/workflow_job_spec.rb]: workflow is tower only feature. Don't need to share.

File: app/models/manageiq/providers/ansible_tower/automation_manager/workflow_job.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      :job_template          => job_template
+    ).tap do |stack|
+      stack.send(:update_with_provider_object, raw_job)
+      stack.update_attributes(:parent => self)
+    end
+  end
+  private :update_child_job
+
+  def update_parameters(raw_job)
+    self.parameters = YAML.safe_load(raw_job.extra_vars).collect do |para_key, para_val|
```
Comment (lfu) [inline app/models/manageiq/providers/ansible_tower/automation_manager/workflow_job.rb]: @jameswnl [Ansible job template job](https://github.com/ManageIQ/manageiq-providers-ansible_tower/blob/ccd0025f8b76fc79d43deffb3eb7ce6452b3ac6a/app/models/manageiq/providers/ansible_tower/shared/automation_manager/job.rb#L104) uses method ```extra_var_hash``` to convert the ```extra_var``` value from YAML format to a hash. Can we add the same method for Ansible workflow template job in to Ansible tower client?

**Comment by bzwei** [comment]
@lfu can we extract common code to a mixin used by both job and workflow_job?

**Comment by lfu** [comment]
Tests passed locally. A new release of Ansible tower client gem might fix it.

**Comment by d-m-u** [comment]
@bzwei the two dependent PRs are merged, what else needs to happen with this one?

**Comment by bzwei** [comment]
@d-m-u LGTM

**Comment by gmcculloug** [comment]
@bzwei Can you look into addressing the rubocop errors.

**Comment by lfu** [comment]
@gmcculloug Rubocop warnings regarding ```:inverse_of``` can't be fixed as the associations here are one way relationship.

**Pull request merged**